src/signal.rs

```diff
@@ -0,0 +1,78 @@
+use anyhow::{Result, anyhow};
+use log::debug;
+use nix::sys::signal::{self, SigHandler, Signal};
+use nix::unistd::Pid;
+use std::sync::atomic::{AtomicI32, Ordering};
+
+/// Set of signals forwarded from the parent to the child
+const FORWARDED_SIGNALS: &[Signal] = &[
+    Signal::SIGTERM,
+    Signal::SIGINT,
+    Signal::SIGQUIT,
+    Signal::SIGHUP,
+    Signal::SIGUSR1,
+    Signal::SIGUSR2,
+];
+
+/// Stored child pid used by the parent signal handler to forward signals
+static CHILD_PID: AtomicI32 = AtomicI32::new(0);
```

Comment:
atomic shouldn’t be needed here, or is rust bullying you into using this?

Reply:
Not really, I thought I was going to have to make this thread safe so I went with atomic and then just left it here. I could use a Cell or something like that